eng/pipelines/templates/steps/build.yml

```diff
@@ -13,32 +13,26 @@ steps:
 
   - pwsh: |
       $folder = "${{parameters.ServiceDirectory}}"
-      if ($folder -eq "*") { $folder = "" }
+      if ($folder -eq "*" -or $folder -eq "auto") { $folder = "" }
       echo "##vso[task.setvariable variable=folder]$folder"
     displayName: "Set folder variable for readme links"
 
+  # we are not passing service directory, so we only ever set dev build to true
   - template: /eng/common/pipelines/templates/steps/daily-dev-build-variable.yml
-    parameters:
-      ServiceDirectory: ${{ parameters.ServiceDirectory }}
 
+  # todo: update set-dev.js to handle multiple service directories based on the ChangedServices value of the diff
+  # currently defaulting to nothing, so we set the entire repo to dev build (we do this for python)
   - script: |
       npm install ./eng/tools/versioning
       node eng/tools/versioning/set-dev.js --build-id "$(Build.BuildNumber)" --repo-root "$(Build.SourcesDirectory)" --service "$(folder)"
       node common/scripts/install-run-rush.js update
     condition: and(succeeded(),eq(variables['SetDevVersion'],'true'))
     displayName: "Update package versions for dev build"
 
-  - task: Powershell@2
-    inputs:
-      filePath: $(Build.SourcesDirectory)/eng/common/scripts/Save-Package-Properties.ps1
-      arguments: >
-        -ServiceDirectory ${{parameters.ServiceDirectory}}
-        -OutDirectory $(Build.ArtifactStagingDirectory)/PackageInfo
-        -AddDevVersion
-      pwsh: true
-      workingDirectory: $(Pipeline.Workspace)
-    displayName: Update package properties with dev version
-    condition: and(succeeded(),eq(variables['SetDevVersion'],'true'))
+  - template: /eng/common/pipelines/templates/steps/save-package-properties.yml
```

Comment:
save-package-properties.yml will call `Save-Package-Properties.ps1` with appropriate `-AddDevVersion` by honoring the `SetDevVersion` set above in `daily-dev-build-variable.yml`